core/libp2p/crypto/key.hpp

```diff
@@ -3,34 +3,59 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#ifndef KAGOME_KEY_HPP
-#define KAGOME_KEY_HPP
+#ifndef KAGOME_LIBP2P_CRYPTO_KEY_HPP
+#define KAGOME_LIBP2P_CRYPTO_KEY_HPP
 
 #include "common/buffer.hpp"
 
 namespace libp2p::crypto {
+
   namespace common {
-    enum class KeyType;
+    enum class KeyType : uint32_t;
   }
   /**
    * Interface for public/private key
    */
   class Key {
+   protected:
+    using Buffer = kagome::common::Buffer;
+
    public:
+    /**
+     * @brief constructor
+     * @param key_type key type
+     * @param bytes key content
+     */
+    Key(common::KeyType key_type, Buffer bytes);
+
+    /**
+     * @brief constructor
+     * @param key_type key type
+     * @param bytes key content
+     */
+//    Key(common::KeyType key_type, Buffer bytes);
+
+    virtual ~Key() = default;
+
     /**
      * Get type of this key
      * @return the type
      */
-    virtual common::KeyType getType() const = 0;
+    virtual common::KeyType getType() const;
 
     /**
      * Get a byte representation of the key
      * @return the bytes
      */
-    virtual const kagome::common::Buffer &getBytes() const = 0;
+    virtual const kagome::common::Buffer &getBytes() const;
 
-    virtual bool operator==(const Key &other) const = 0;
+    virtual bool operator==(const Key &other) const;
```

Comment:
comparison operator can be a free function

https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c161-use-nonmember-functions-for-symmetric-operators

Reply:
Ok, made it free function.